### PR TITLE
Do not store park name as user string

### PR DESCRIPTION
**Pull request by IntelOrca**
When looking at the new save format, I have come to the conclusion that it is not necessary to store strings in a user table. Strings should be stored on the objects they relate to as `std::string`.

Starting with park name, I have moved it to be stored as a simple `std::string` in the `Park` class. When exporting back to S6, it allocates the user string for backwards compatibility. In rare the occurrence all user strings are used up, the park name gets saved as "unnamed park".

The benefit now is that the park name can share the same name as a ride.

### CODE REVIEW

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 1]
```diff
@@ -406,15 +407,14 @@ static bool browse(bool isSave, char* path, size_t pathSize)
...
-            // Use localized "Unnamed Park" if park name was empty
-            if (String::SizeOf(buffer) == 0)
-                format_string(buffer, pathSize, STR_UNNAMED_PARK, nullptr);
-
-            safe_strcat_path(path, buffer, pathSize);
+            auto& park = OpenRCT2::GetContext()->GetGameState()->GetPark();
+            auto buffer = park.Name;
+            if (buffer.empty())
+            {
+                // Use localized "Unnamed Park" if park name was empty
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/LoadSave.cpp]:
```suggestion
                // Use localised "Unnamed Park" if park name was empty.
```

(Might as well... :P)

File: src/openrct2/world/Park.cpp [hunk 2]
```diff
@@ -79,17 +77,6 @@ int32_t _suggestedGuestMaximum;
  */
 int32_t _guestGenerationProbability;
 
-/**
- *
- *  rct2: 0x00667104
- */
-void reset_park_entry()
```
Comment (Gymnasiast) [inline src/openrct2/world/Park.cpp]: Why did you remove this?

Comment (IntelOrca) [reply src/openrct2/world/Park.cpp]: It was only called from one place which already initialised the data.

File: src/openrct2/drawing/ScrollingText.cpp [hunk 3]
```diff
@@ -19,20 +19,15 @@
 #include <algorithm>
 #include <mutex>
 
-#pragma pack(push, 1)
-/* size: 0xA12 */
 struct rct_draw_scroll_text
```
Comment (Gymnasiast) [inline src/openrct2/drawing/ScrollingText.cpp]: I take it this structure is no longer tied to SV6 (or wasn't in the first place)? In that case, it's perhaps better to drop the `rct_` prefix and use TitleCase.

Comment (IntelOrca) [reply src/openrct2/drawing/ScrollingText.cpp]: It was never tied, just a draw structure.

File: src/openrct2-ui/windows/Park.cpp [hunk 4]
```diff
@@ -582,8 +583,11 @@ static void window_park_set_disabled_tabs(rct_window* w)
 
 static void window_park_prepare_window_title_text()
 {
-    set_format_arg(0, rct_string_id, gParkName);
-    set_format_arg(2, uint32_t, gParkNameArgs);
+    auto& park = OpenRCT2::GetContext()->GetGameState()->GetPark();
+    auto parkName = park.Name.c_str();
+
+    set_format_arg(0, rct_string_id, STR_STRING);
+    set_format_arg(2, const char*, parkName);
```
Comment (Broxzier) [inline src/openrct2-ui/windows/Park.cpp]: This may be what causes some of the backtrace reports. `parkName` points to a temporary buffer returned by `c_str` that is not guaranteed to exist after this function call. Forcing this function to be inlined and calling it just before `gCommonFormatArgs` gets used may fix it.

Comment (IntelOrca) [reply src/openrct2-ui/windows/Park.cpp]: Thanks for spotting this, I will try to resolve...

Comment (IntelOrca) [reply src/openrct2-ui/windows/Park.cpp]: On closer inspection of the disassembly, this is not a temporary. It is actually using the internal buffer of the std::string on the global Park instance. The std::string is not copied.

Comment (IntelOrca) [reply src/openrct2-ui/windows/Park.cpp]: Okay, I found the cause. It is because the price of the park is overwriting the park name args in the arg buffer.

`set_format_arg(6, uint32_t, parkEntranceFee);`
